hw/ip/sram_ctrl/data/sram_ctrl.hjson

```diff
@@ -106,159 +108,133 @@
+        swaccess: "rw0c",
+        hwaccess: "none",
+        fields: [
+          { bits: "0"
+            desc: '''
+                  When cleared to zero, !!EXEC can not be written anymore.
+                  ''',
+            resval: 1
+          }
+        ]
+      }
+      { name: "EXEC",
+        desc: "Sram execution enable.",
+        swaccess: "rw",
+        hwaccess: "hro",
+        regwen:   "EXEC_REGWEN"
+        fields: [
+          { bits: "2:0",
+            name: "EN",
+            desc: '''
+                    Write 5 (3b101) to this field to enable execution from SRAM
+                  '''
+          },
+        ]
+      },
+      { name: "CTRL_REGWEN",
+        desc: "Lock register for control register.",
+        swaccess: "rw0c",
+        hwaccess: "none",
+        fields: [
+          { bits: "0"
+            desc: '''
+                  When cleared to zero, !!CTRL can not be written anymore.
+                  ''',
+            resval: 1
+          }
+        ]
+      }
+      { name: "CTRL",
+        desc: "SRAM ctrl register.",
+        swaccess: "rw",
+        hwaccess: "hrw",
+        hwqe:     "true",
+        hwext:    "true",
+        regwen:   "CTRL_REGWEN"
+        tags: [// avoid writing to CTRL, as this will cause STATUS to be modified
+               "excl:CsrNonInitTests:CsrExclWrite"]
+        fields: [
+          { bits: "0",
+            name: "RENEW_SCR_KEY",
+            desc: '''
+                  Write 1 to request a new scrambling key from OTP. After writing to this register, SRAM transactions will
+                  be blocked until !!STATUS.SCR_KEY_VALID has been set to 1. If !!STATUS.SCR_KEY_VALID was already 1
+                  before triggering a key renewal, hardware will automatically clear that status bit such that software
+                  can poll its status. Note that requesting a new scrambling key takes ~200 OTP cycles, which translates
+                  to ~800 CPU cycles (OTP runs at 24MHz, CPU runs at 100MHz). Note that writing 1 to this register while
+                  a key request is pending has no effect.
+                  '''
+          },
+          { bits: "1",
+            name: "INIT",
+            desc: '''
+                  Write 1 to request memory init.
+                  The init process is seeded using the a nonce that is supplied during the init process.
+                  The resulting memory is programmed with random data instead of deterministic values.
+                  '''
+          },
+        ]
+      },
+    ],
 
-    { name: "ERROR_ADDRESS",
-      desc: "Error address register.",
-      swaccess: "ro",
-      hwaccess: "hwo",
-      hwqe:     "false",
-      hwext:    "false",
-      fields: [
-        { bits: "31:0"
-          name: "ADDRESS"
-          desc: '''
-                Contains the byte address of the last parity error.
-                '''
-        }
-      ]
-    }
-  ],
+    ram: [
+      // no CSRs defined here.
```

Comment:
In top_earlgrey.hjson, we define the memory as below, but it doesn't have the completed info, for example, `byte_write`. Is it better if we have all the mem info in the block and only override what we need to change in the top-level, so that IP hjson is also self-contained.
      memory: {
        ram: {
          label:    "ram_ret_aon", // label for linker script
          swaccess: "rwx",         // access attributes for linker script
          size:     "0x1000"       // size in bytes for linker script, xbar and RTL parameterization
        }
      }

Reply:
Hey @weicaiyang, I think ideally we had a mechanism in place to override the parameterization of windows in the block Hjson - but I believe this is not very straightforward to do, since the CSR tops are generated statically from the Hjson.

So for now, I think it would be OK to add the missing information to the dict in the toplevel. Is it only `byte-write` that was missing?

Reply:
@weicaiyang I have added the missing keys and aligned the swaccess labels in #7507

Reply:
thanks Michael. I have updated #7504 and ran it with this PR which is passing on chip-level smoke test.

Reply:
Thanks for checking with this PR, and sorry for my slow response - I got tied up with something else.
I will have a look at #7504 now.